airflow/contrib/hooks/databricks_hook.py

```diff
@@ -175,6 +190,12 @@ def cancel_run(self, run_id):
         self._do_api_call(CANCEL_RUN_ENDPOINT, json)
 
 
+def _retryable_error(exception):
+    return type(exception) == requests_exceptions.ConnectionError \
```

Comment:
@andrewmchen Oh, that is completely right! Otherwise derived exceptions won't be treated as retryable. Thanks for spotting that. I did a small change to the tests to cover this case too.